diff --git a/src/cpu/o3/commit.cc b/src/cpu/o3/commit.cc
index 815d9497ff..d5ebf95243 100644
--- a/src/cpu/o3/commit.cc
+++ b/src/cpu/o3/commit.cc
@@ -154,6 +154,9 @@ Commit::CommitStats::CommitStats(CPU *cpu, Commit *commit)
                "communicate backwards"),
       ADD_STAT(branchMispredicts, statistics::units::Count::get(),
                "The number of times a branch was mispredicted"),
+      ADD_STAT(branchMispredictsCond, statistics::units::Count::get(),
+               "The number of times a conditional branch was "
+               "mispredicted"),
       ADD_STAT(numCommittedDist, statistics::units::Count::get(),
                "Number of insts commited each cycle"),
       ADD_STAT(amos, statistics::units::Count::get(),
@@ -812,7 +815,9 @@ Commit::commit()
                                     rob->findInst(tid, squashed_inst);
             if (toIEW->commitInfo[tid].mispredictInst) {
                 if (toIEW->commitInfo[tid].mispredictInst->isUncondCtrl()) {
-                     toIEW->commitInfo[tid].branchTaken = true;
+                    toIEW->commitInfo[tid].branchTaken = true;
+                } else if (toIEW->commitInfo[tid].mispredictInst->isCondCtrl()) {
+                    ++stats.branchMispredictsCond;
                 }
                 ++stats.branchMispredicts;
             }
diff --git a/src/cpu/o3/commit.hh b/src/cpu/o3/commit.hh
index 4fff9fe892..509acbb554 100644
--- a/src/cpu/o3/commit.hh
+++ b/src/cpu/o3/commit.hh
@@ -476,6 +476,11 @@ class Commit
          * squash.
          */
         statistics::Scalar branchMispredicts;
+        /**
+         * Extension of the above statistic, only counting conditional
+         * mispredicts.
+         */
+        statistics::Scalar branchMispredictsCond;
         /** Distribution of the number of committed instructions each cycle. */
         statistics::Distribution numCommittedDist;
 
